src/renderer/components/+landing-page/workspace-cluster.store.ts

```diff
@@ -0,0 +1,64 @@
+import { WorkspaceId } from "../../../common/workspace-store";
+import { Cluster } from "../../../main/cluster";
+import { clusterStore } from "../../../common/cluster-store";
+import { ItemObject, ItemStore } from "../../item.store";
+import { autobind } from "../../utils";
+
+export class ClusterItem implements ItemObject {
+  cluster: Cluster;
+
+  getName() {
+    return this.cluster.name;
+  }
+
+  getId() {
+    return this.cluster.id;
+  }
+}
+
+/** an ItemStore of the clusters belonging to a given workspace */
+@autobind()
+export class WorkspaceClusterStore extends ItemStore<ClusterItem> {
+
+  workspaceId: WorkspaceId;
+
+  constructor(workspaceId: WorkspaceId) {
+    super();
+    this.workspaceId = workspaceId;
+  }
+
+  loadAll() {
+    return this.loadItems(() => clusterStore.getByWorkspaceId(this.workspaceId).map(cluster => {
+      const clusterItem = new ClusterItem();
+
+      clusterItem.cluster = cluster;
+
+      return clusterItem;
+    }));
+  }
+
+  async remove(clusterItem: ClusterItem) {
+    const { cluster } = clusterItem;
+
+    if (cluster.isManaged) {
+      return;
+    }
+
+    const clusterId = cluster.id;
+
+    return super.removeItem(clusterItem, async () => {
```

Comment:
I don't think this function needs to be async
```suggestion
    return super.removeItem(clusterItem, () => {
```

Reply:
It has to return a `Promise` so specifying `async` is typical, right?

Reply:
I don't think so, one can return promise without `async`. However, an `async` function always wraps its return value in a promise. 

Furthermore, you are not actually calling any other functions that return promises. Finally, async functions always add their "body" to the event loop.